pkg/services/govmomi/vcenter/clone.go

```diff
@@ -138,10 +142,11 @@ func Clone(ctx context.Context, vmCtx *capvcontext.VMContext, bootstrapData []by
 
 	// Only non-linked clones may expand the size of the template's disk.
 	if snapshotRef == nil {
-		diskSpecs, err := getDiskSpec(vmCtx, devices)
+		diskSpecs, err := getDiskSpec(ctx, vmCtx, devices)
```

Comment:
This only runs on non-linked clones.

But adding disks may be fine for linked clones too?!

Othwerwise: if this only works for non-linked clones, we should catch that in a validation webhook to ensure linkedClone is not set. (.spec.cloneMode)

Reply:
Refactored code to allow for all clone types.  Seems to be working now for PR e2e as well.  Thanks!